crates/aiken-project/src/blueprint/validator.rs

```diff
@@ -95,6 +125,25 @@ impl Validator<Schema> {
     }
 }
 
+impl<T> Validator<T>
+where
+    T: Clone,
+{
+    pub fn apply(self, arg: &Term<DeBruijn>) -> Result<Self, Error> {
+        match self.parameters.split_first() {
+            None => Err(Error::NoParametersToApply),
+            Some((_, tail)) => {
+                // TODO: Ideally, we should control that the applied term matches its schema.
```

Comment:
This requires providing `Deserialize` instance for `Schema` and `Data` and having some kind of `Term` validation against a Schema. I'll work on that in another increment.